backend/dal/Helpers/PimsMigrationSqlGenerator.cs

```diff
@@ -56,113 +47,44 @@ public PimsMigrationSqlGenerator(MigrationsSqlGeneratorDependencies dependencies
         /// Get the table name abbreviation for the specified table.
         /// </summary>
         /// <param name="table"></param>
-        /// <returns></returns>
+        /// <returns>null if there is no match or the entity has been deleted in a future migration</returns>
         string GetAbbreviation(string table)
         {
             return table switch
             {
                 "__EFMigrationsHistory" => "MIGHST",
                 "PIMS_MIGRATION_HISTORY" => "MIGHST",
-                _ => _tables.GetValueOrDefault(table) ?? throw new InvalidOperationException($"Abbreviation is required for table {table}")
+                _ => _tables.GetValueOrDefault(table)
             };
         }
 
-        /// <summary>
-        /// Get the entity type for the specified table name.
-        /// </summary>
-        /// <param name="table"></param>
-        /// <returns></returns>
-        Type GetEntityType(string table)
+        protected override void Generate(CreateTableOperation operation, IModel model, MigrationCommandListBuilder builder, bool terminate = true)
         {
-            return table switch
+            Dictionary<string, string> defaultValues = new Dictionary<string, string>();
+            foreach (var column in operation.Columns)
             {
-                "__EFMigrationsHistory" => null,
-                "PIMS_MIGRATION_HISTORY" => null,
-                _ => _tableTypes.GetValueOrDefault(table) ?? throw new InvalidOperationException($"Entity model is required for table {table}")
-            };
-        }
-
-        /// <summary>
-        /// Extract the table and column name for subsequent steps.
-        /// </summary>
-        /// <param name="schema"></param>
-        /// <param name="table"></param>
-        /// <param name="name"></param>
-        /// <param name="operation"></param>
-        /// <param name="model"></param>
-        /// <param name="builder"></param>
-        protected override void ColumnDefinition(
-            string schema,
-            string table,
-            string name,
-            ColumnOperation operation,
-            IModel model,
-            MigrationCommandListBuilder builder)
-        {
-            _currentEntityType = GetEntityType(table);
-            _currentTableName = table;
```

Comment:
This isn't thread safe, so there may be race conditions where the _currentTableName gets used innapropriately.